### PR TITLE
Fix info leak when using a remote daemon

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -2784,6 +2790,7 @@ void wallet2::pull_and_parse_next_blocks(uint64_t start_height, uint64_t &blocks
       }
     }
     waiter.wait(&tpool);
+    last = !blocks.empty() && cryptonote::get_block_height(parsed_blocks.back().block) + 1 == blocks_start_height + parsed_blocks.size();
```
Comment (xiphon) [inline src/wallet/wallet2.cpp]: Isn't this always true (given that `blocks` is not empty)?
```cryptonote::get_block_height(parsed_blocks.back().block) + 1 == blocks_start_height + parsed_blocks.size()```

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: It should be if the sender sends correct data.

Comment (xiphon) [reply src/wallet/wallet2.cpp]: But will `refresh` handle fetching 2500 blocks (it needs 3 `get_blocks` calls)?
Seems the first `pull_and_parse_next_blocks` invocation will set `last` to `true`.

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: Pretty sure I tried that. I'll check.

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: Looks like I lost some patch, I was using the returned chain height. Fixing, you are right,

**Pull request merged**